smoderp2d/processes/infiltration/__init__.py

```diff
@@ -59,3 +55,11 @@ def phlilip(k, s, deltaT, totalT, NoDataValue):
 
         # except ValueError:
     return infiltration
+
+def philip_implicit(k, s, deltaT, totalT, NoDataValue):
+    if k and s == NoDataValue:
+        infiltration = NoDataValue
+    else:
+        infiltration = (0.5 * ma.divide(s, ma.sqrt(totalT + deltaT)) + k) * \
+                deltaT
```

Comment:
Comparing the new code with the original philip - why do we not need to do this `ma.where` anymore? What is the explanation of these changes purpose?

Reply:
Corrected